sessionctx/stmtctx/stmtctx.go

```diff
@@ -243,6 +243,11 @@ type StatementContext struct {
 
 	// RCCheckTS indicates the current read-consistency read select statement will use `RCCheckTS` path.
 	RCCheckTS bool
+
+	// IsSQLRegistered uses to indicate whether the SQL has been registered for TopSQL.
+	IsSQLRegistered atomic2.Bool
+	// IsSQLAndPlanRegistered uses to indicate whether the SQL and plan has been registered for TopSQL.
+	IsSQLAndPlanRegistered atomic2.Bool
```

Comment:
Will it happen that `IsSQLAndPlanRegistered == true` while `IsSQLRegistered == false`?

Reply:
It will happen, but it doesn't matter, the code can make sure if `IsSQLAndPlanRegistered ` is true, then the `SQL` and `plan` must be registerd.